src/data/polynomial/degree/basic.lean

```diff
@@ -268,6 +271,54 @@ by simpa only [C_1, one_mul, pow_one] using degree_C_mul_X_pow_le (1:R) 1
 lemma nat_degree_X_le : (X : polynomial R).nat_degree ≤ 1 :=
 nat_degree_le_of_degree_le degree_X_le
 
+lemma mem_support_C_mul_X_pow {n a : ℕ} {c : R} : a ∈ (C c * X ^ n).support → a = n :=
+begin
+  rw [mem_support_iff_coeff_ne_zero, coeff_C_mul_X c n a],
+  split_ifs,
+    { intro,
+      assumption, },
+    { intro a_1,
+    exfalso,
+    apply a_1,
+    refl, },
```

Comment:
```suggestion
  { intro,
    assumption, },
  { intro a_1,
    exfalso,
    apply a_1,
    refl, },
```

Reply:
The variable name `a_1` is "evil" since it was auto-generated. Please use `split_ifs with foobar` to generate a name `foobar` instead. Good names are probably `h` or `ha` or `a0` or `ha0` or something like that.

Reply:
I change ```split_ifs```to ```split_ifs with h```, but it is still the hypothesis called ```a_1``` that I explicitly apply.  The ```split_ifs``` hypothesis is hidden in ```assumption```.  Do you still recommend defining it explicitly, or is it ok to use ```split_ifs```?